tests/ssl-opt.sh

```diff
@@ -3976,24 +4004,14 @@ run_test    "Authentication: client no cert, openssl server required" \
             -c "skip write certificate verify" \
             -c "! mbedtls_ssl_handshake returned"
 
-# The "max_int chain" tests assume that MAX_INTERMEDIATE_CA is set to its
-# default value (8)
+# config.h contains a value for MBEDTLS_X509_MAX_INTERMEDIATE_CA that is
+# different from the script's assumed default value (below).
+# Relevant tests are skipped if they do not match.
 
 MAX_IM_CA='8'
-MAX_IM_CA_CONFIG=$( ../scripts/config.py get MBEDTLS_X509_MAX_INTERMEDIATE_CA)
-
-if [ -n "$MAX_IM_CA_CONFIG" ] && [ "$MAX_IM_CA_CONFIG" -ne "$MAX_IM_CA" ]; then
-    cat <<EOF
-${CONFIG_H} contains a value for the configuration of
-MBEDTLS_X509_MAX_INTERMEDIATE_CA that is different from the script's
-test value of ${MAX_IM_CA}.
-
-The tests assume this value and if it changes, the tests in this
-script should also be adjusted.
-EOF
-    exit 1
-fi
 
+requires_config_value_at_least "MBEDTLS_X509_MAX_INTERMEDIATE_CA" $MAX_IM_CA
+requires_config_value_at_most "MBEDTLS_X509_MAX_INTERMEDIATE_CA" $MAX_IM_CA
 requires_full_size_output_buffer
 run_test    "Authentication: server max_int chain, client default" \
             "$P_SRV crt_file=data_files/dir-maxpath/c09.pem \
```

Comment:
I think we could make the tests adapt to values of `MBEDTLS_X509_MAX_INTERMEDIATE_CA` between 2 (or whatever the minimum is to pass other test cases — we don't care about insanely low values breaking the script) and 18 by using `$(printf c%02d $((MBEDTLS_X509_MAX_INTERMEDIATE_CA+1))).pem` (or a faster implementation of that) instead of `c09.pem`, etc. I'm not sure it's worth the effort though.

Reply:
I thought about this too, but I don't think it's worth the effort. It's not something essential as far as I see, and I don't see how different values could reveal more bugs, unless the library does something different. Even if it is worth, I think it should be a separate PR because this one is only for the replacement of hard-aborts. I heard PRs should be small and fix one thing! :)